src/Control/SimpleResourceURLGenerator.php

```diff
@@ -194,6 +193,9 @@ protected function resolvePublicResource($relativePath)
         if (!$publicOnly && file_exists($privatePath ?? '')) {
             // String is private but exposed to _resources/, so rewrite to the symlinked base
             $relativePath = Path::join(RESOURCES_DIR, $relativePath);
+            if (!Path::isBasePath(RESOURCES_DIR, $privatePath)) {
+                throw new InvalidArgumentException("'$relativePath' must not be outside the resources root");
+            }
```

Comment:
Added traversal protection here because it seems likely for the return value to be directly used to read files at the returned path, and we have no idea where the relative path is coming from.